### PR TITLE
Rename sd to sigma for consistency.

### PR DESCRIPTION
**Pull request by twiecki**
A long time ago I made the terrible choice of naming the scale parameter of the Normal distribution `sd` instead of `sigma`.  This is a terrible choice because all our other parameter names follow greek names, so it should either be `mean` and `sd`, or `mu` and `sigma`. Since we're using greek everywhere else it should be the same for `sd`, I mean `sigma`...

This was also rightfully criticized by @bob-carpenter in his blog post https://andrewgelman.com/2017/05/31/compare-stan-pymc3-edward-hello-world/
image

This so far replaces sd->sigma everywhere in the `continuous.py` but also allows passing `sd` to keep backwards compatibility. I don't think we should really deprecate this because *everyone* is using it but allow for a more consistent naming to emerge.

If we decide to go ahead we should probably also fix this in the docs, so I marked this as WIP.

### CODE REVIEW

File: pymc3/distributions/mixture.py [hunk 1]
```diff
@@ -217,27 +217,30 @@ class NormalMixture(Mixture):
...
 
-    Note: You only have to pass in sd or tau, but not both.
+    Note: You only have to pass in sigma or tau, but not both.
     """
 
     def __init__(self, w, mu, comp_shape=(), *args, **kwargs):
-        _, sd = get_tau_sd(tau=kwargs.pop('tau', None),
-                           sd=kwargs.pop('sd', None))
+        if 'sd' in kwargs.keys():
+            kwargs['sigma'] = kwargs.pop('sd')
```
Comment (ferrine) [inline pymc3/distributions/mixture.py:224]: should it warn about deprecation?

File: pymc3/distributions/dist_math.py [hunk 2]
```diff
@@ -108,17 +108,17 @@ def normal_lccdf(mu, sigma, x):
...
-    `sd -> rho` theano converter
-    :math:`mu + sd*e = mu + log(1+exp(rho))*e`"""
-    return tt.log(tt.exp(tt.abs_(sd)) - 1.)
+    `sigma -> rho` theano converter
+    :math:`mu + sigma*e = mu + log(1+exp(rho))*e`"""
+    return tt.log(tt.exp(tt.abs_(sigma)) - 1.)
 
 
-def rho2sd(rho):
+def rho2sigma(rho):
```
Comment (ferrine) [inline pymc3/distributions/dist_math.py:116]: Do not forget to run
https://github.com/pymc-devs/pymc3/blob/c496378db707f402c89add9b00ea70015ecab50c/docs/source/notebooks/convolutional_vae_keras_advi.ipynb

Comment (ferrine) [reply pymc3/distributions/dist_math.py:116]: Also a backward compat function warning about deprecation is needed

**Comment by ferrine** [comment]
When should we drop support for old convention?
* never? Just warn users to use proper convention
* on next 3.7 release? 3.6 should warn I suppose

File: pymc3/distributions/dist_math.py (hunk 2 above)
Comment (junpenglao) [reply pymc3/distributions/dist_math.py:116]: +1 We should have rho2d = rho2sigma

**Comment by twiecki** [comment]
I'm not sure we should ever drop support, but high uncertainty on what the best course of action is. If we strongly feel that this renaming is the right way forward, we should raise a deprecation warning in  the next release and drop it after.

**Comment by fonnesbeck** [comment]
I agree that adopting Greek as a package-wide convention is a good idea. I don't think there's any harm in allowing `sd` to be specified in perpetuity for the sake of compatibility.

**Comment by twiecki** [comment]
Docs are updated and made a comment in the release notes.

**Pull request merged**